### PR TITLE
Size metrics for lucene file reference write and count metrics for lu…

### PR DESCRIPTION
**Pull request by tian-yizuo**
…cene index write

### CODE REVIEW

File: fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryTest.java [hunk 1]
```diff
@@ -92,7 +92,8 @@ public void testWriteLuceneFileReference() throws Exception {
         CompletableFuture<FDBLuceneFileReference> luceneFileReference = directory.getFDBLuceneFileReference("test1");
         assertNotNull(luceneFileReference.get(5, TimeUnit.SECONDS), "fileReference should exist");
 
-        assertCorrectMetricCount(FDBStoreTimer.Counts.LUCENE_WRITE_FILE_REFERENCE,2);
+        assertCorrectMetricCount(FDBStoreTimer.Counts.LUCENE_WRITE_FILE_REFERENCE_SIZE, reference1.getBytes().length + reference2.getBytes().length);
+        assertCorrectMetricCount(FDBStoreTimer.Counts.LUCENE_WRITE_FILE_REFERENCE_CALL,2);
```
Comment (MMcM) [inline fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryTest.java:96]:
```suggestion
        assertCorrectMetricCount(FDBStoreTimer.Counts.LUCENE_WRITE_FILE_REFERENCE_CALL, 2);
```

File: fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java [hunk 2]
```diff
@@ -254,11 +263,9 @@ public void writeFDBLuceneFileReference(@Nonnull String name, @Nonnull FDBLucene
      * @param value the data to be stored
      */
     public void writeData(long id, int block, @Nonnull byte[] value) {
-        FDBStoreTimer timer = context.getTimer();
-        if (timer != null) {
-            //This may not be correct transactionally
-            timer.increment(FDBStoreTimer.Counts.LUCENE_WRITE_SIZE, value.length);
-        }
+        //This may not be correct transactionally
```
Comment (MMcM) [inline fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java:266]: I am not sure what potential problem this comment is referring to.

File: fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java [hunk 3]
```diff
@@ -164,6 +164,13 @@ private void incrementCallCount(FDBStoreTimer.Counts counter) {
         }
     }
 
+    private void incrementSize(FDBStoreTimer.Counts counter, int size) {
```
Comment (MMcM) [inline fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java:167]: Why have the more complicated contract that this checks whether counter is a size and does nothing? Why not just another overload of `increment` that takes an `amount`?

**Review by MMcM — CHANGES_REQUESTED** [review]

File: fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java (hunk 2 above)
Comment (tian-yizuo) [reply fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java:266]: I am also not really sure. I just keep the existing comment. Probably @scottfines may have better idea?

File: fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java (hunk 3 above)
Comment (tian-yizuo) [reply fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java]: My understanding is that we use the amount > 1 only for incrementing data size (although the underlying ```StoreTimer.increment(@Nonnull Count event, int amount)``` is not designed to be specific to data size). And we can see the method above has a specific name ```incrementCallCount```, which looks like specific for incrementing of calling count. If so, checker for ```counter.isSize()``` may be meaningful here. But I just looked at the similar code in ```InstrumentedReadTransaction``` and see it just simply have 2 overloads for ```increment```, one with amount argument and the other is to increment only 1. So I just changed the 2 methods here in same way. Please let me know if it looks good.

File: fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryTest.java (hunk 1 above)
Comment (tian-yizuo) [reply fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryTest.java]: Fixed.

**Review by MMcM — APPROVED** [review]

File: fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java (hunk 2 above)
Comment (scottfines) [reply fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java:266]: it's not a big deal, I was referring to the possibility that if you rollback after incrementing this timer, the timer will technically be incorrect. This code doesn't do anything that affects the comment one way or the other.

**Review by scottfines — APPROVED** [review]

**Review by jleach4 — APPROVED** [review]

**Pull request merged**